File: E-104-maximum-depth-of-binary-tree/solution.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function (root, dept = 0) {
  if (!root) return dept;
  const newDept = dept + 1;
  return Math.max(maxDepth(root.left, newDept), maxDepth(root.right, newDept));
};

File: E-104-maximum-depth-of-binary-tree/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root, dept=0):
        if root is None:
            return dept

        return max(self.maxDepth(root.left, dept + 1), self.maxDepth(root.right, dept + 1))
